ruby/ql/src/queries/modeling/internal/Summaries.qll

```diff
@@ -0,0 +1,73 @@
+/**
+ * Contains predicates for generating `summaryModel`s to summarize flow through methods.
+ */
+
+private import ruby
+private import codeql.ruby.ApiGraphs
+private import codeql.ruby.TaintTracking
+private import Util as Util
+
+/**
+ * Contains predicates for generating `summaryModel`s to summarize flow through methods.
+ */
+module Summaries {
+  private module Config implements DataFlow::ConfigSig {
+    predicate isSource(DataFlow::Node source) { source instanceof DataFlow::ParameterNode }
+
+    predicate isSink(DataFlow::Node sink) { sink = any(DataFlow::MethodNode m).getAReturnNode() }
+  }
+
+  API::Node getAnyParameterNode(DataFlow::MethodNode methodNode) {
+    result.asSource() =
+      [
+        methodNode.getParameter(_), methodNode.getKeywordParameter(_),
+        methodNode.getBlockParameter(), methodNode.getSelfParameter()
+      ]
+  }
+
+  private module ValueFlow {
+    import DataFlow::Global<Config>
+
+    predicate summaryModel(string type, string path, string input, string output) {
+      exists(DataFlow::MethodNode methodNode, API::Node paramNode |
+        methodNode.getLocation().getFile() instanceof Util::RelevantFile and
+        paramNode.getAValueReachableFromSource() = methodNode.getAReturnNode() and
```

Comment:
```suggestion
        flow(paramNode.asSource(), methodNode.getAReturnNode()) and
```
Any reason we don't use `flow` here, as below?

Reply:
Nope, I was trying this out to see if there was any difference between them and there didn't appear to be. `flow` is nicer for consistency with `TaintFlow`.